dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java

```diff
@@ -45,7 +45,8 @@ protected String rewriteTarget(HttpServletRequest clientRequest) {
     String targetAddress =
         coordinatorServerAddressesMap.get(clientRequest.getHeader("targetAddress"));
     if (targetAddress == null) {
-      return null;
+      // Get random one from coordinatorServerAddressesMap
+      targetAddress = coordinatorServerAddressesMap.values().iterator().next();
```

Comment:
Will `coordinatorServerAddressesMap.values()` be empty?

Reply:
I think if `coordinatorServerAddressesMap` is empty, it should throw an exception when starts

Reply:
yes, it will throw another NPE....

```Console
[2024-07-03 11:35:34.514] [main] [INFO] JettyServerFront.main - Start to init dashboard http server using config /Users/mbl/projects/tencent/incubator-uniffle/conf/dashboard.conf
[2024-07-03 11:35:34.562] [main] [INFO] RssUtils.getPropertiesFromFile - Load config from /Users/mbl/projects/tencent/incubator-uniffle/conf/dashboard.conf
[2024-07-03 11:35:34.587] [main] [INFO] log.initialized - Logging initialized @1726ms
Exception in thread "main" java.lang.NullPointerException
	at org.apache.uniffle.dashboard.web.utils.DashboardUtils.convertAddressesStrToMap(DashboardUtils.java:34)
	at org.apache.uniffle.dashboard.web.JettyServerFront.setRootServletHandler(JettyServerFront.java:96)
	at org.apache.uniffle.dashboard.web.JettyServerFront.initialization(JettyServerFront.java:86)
	at org.apache.uniffle.dashboard.web.JettyServerFront.<init>(JettyServerFront.java:63)
	at org.apache.uniffle.dashboard.web.JettyServerFront.main(JettyServerFront.java:75)
```

Reply:
@xianjingfeng I push a new commit to add a precondition check to avoid NPE. PTAL.

Reply:
I mean throw an exception if `coordinatorServerAddressesMap.isEmpty() == true`

Reply:
> I mean throw an exception if `coordinatorServerAddressesMap.isEmpty() == true` when the process starts.

@maobaolong @rickyma

Reply:
Maybe: 
```
Preconditions.checkArgument(coordinatorServerAddressesMap.isEmpty(), "No coordinator server address found.");
```